theories/distributions/Distr.ec

```diff
@@ -693,6 +767,19 @@ proof. by move=> x1 x2;rewrite !supp_dunit => -> ->. qed.
 
 lemma finite_dunit ['a] (x : 'a) : is_finite (support (dunit x)).
 proof. by apply/uniform_finite/dunit_uni. qed.
+
+lemma dunit_pmax (x: 'a) :
+  p_max (dunit x) = 1%r.
+proof.
+apply eqr_le; split => [|_]; first apply flub_le_ub => ?; by apply le1_mu.
+have ->: 1%r = mu1 (dunit x) x by rewrite /= dunit1E.
```

Comment:
Because the lemma statement looks like `lemma dunit1E ['a]: forall (x y : 'a), mu1 (dunit x) y = b2r (x = y).`

I'm not sure how to match my `1` against `b2r (x = x)` in a simpler way.

Reply:
Ah. Option 1 (will work):

`have @/b2r /= <- := dunit1E x x.`

(Instantiate `dunit1E` with `x` and `x`, make it the top assumption, then unfold `b2r`, simplify `b2r (x = x)` and rewrite right to left.)

Option 2 (may work): try `rewrite -/(b2r (x = x))`.

Reply:
Option 2 doesn't work. "not headed by an operator/predicate".

Reply:
Option 1 works as expected. Never seen the `:=` syntax before; I need to play with it sometime